src/main/scala/org/apache/spark/sql/redis/RedisSourceRelation.scala

```diff
@@ -59,7 +59,9 @@ class RedisSourceRelation(override val sqlContext: SQLContext,
   /** parameters **/
   private val tableNameOpt: Option[String] = parameters.get(SqlOptionTableName)
   private val keysPatternOpt: Option[String] = parameters.get(SqlOptionKeysPattern)
+  private val keysPrefixPattern = keysPatternOpt.orElse(tableNameOpt).getOrElse("")
```

Comment:
There is a dataKeyPattern() function that basically does the same. Also, I would rather use Option that an empty string.

Reply:
agree